### ISSUE TITLE
"Do move_base to ChargingPoint"

### ISSUE THREAD

**Issue by hawesie** [strands-project/strands_navigation#208]
This is in the system running on Bob right now. Bob is docked and charging, `/current_node` states 'ChargingPoint'. Due to a bug (https://github.com/strands-project/strands_executive_behaviours/issues/19) the routine is trying to send the robot to 'ChargingPoint'.  This appears to be triggering the nav system to do a movebase move to the ChargingPoint (`Do move_base to ChargingPoint` is appearing at the start of a failed nav move), and this is failing repeatedly (of course). 

This is running the code from #207 with the mdp executor https://github.com/strands-project/strands_executive/pull/142.

**Comment by Jailander** [comment]
This issue is a duplicate of #160 is going to be solved by using (https://github.com/strands-project/strands_navigation/blob/indigo-devel/strands_navigation_msgs/msg/TopologicalNode.msg#L9)

**Comment by hawesie** [comment]
Closed as a duplicate of #160.

**Comment by bfalacerda** [comment]
There is some extra issue here, when docking is successful it does the "Do move_base to ChargingPoint" thing immediately after. Here is some output. 

```
 INFO] [1427740494.912591915]: Adjusting position: -0.049580 -0.070520
[ INFO] [1427740494.942854668]: Charging service is measuring robot position
[ INFO] [1427740499.414260020]: Charging service is attempting to dock
[INFO] [WallTime: 1427740503.177123] new node reached ChargingPoint
[INFO] [WallTime: 1427740503.177768] new node reached ChargingPoint
[ INFO] [1427740505.160004776]: Charging service is waiting for charger signal
[ INFO] [1427740510.091284774]: Position of the robot relative to the charging station is -0.002 -0.003 -0.001, i.e. 4 mm
 from the desired position->                                                                                             
[ INFO] [1427740510.124363073]: Charging service is feeding
[INFO] [WallTime: 1427740510.178746] State machine terminating 'NAVIGATION':'succeeded':'succeeded'
[INFO] [WallTime: 1427740510.179119] Concurrent state 'NAV_SM' returned outcome 'succeeded' on termination.
[INFO] [WallTime: 1427740510.183524] Concurrent state 'stuck_on_carpet' returned outcome 'preempted' on termination.
[INFO] [WallTime: 1427740510.192023] Concurrent state 'bumper' returned outcome 'preempted' on termination.
[INFO] [WallTime: 1427740510.195622] Concurrent state 'service_pause_nav' returned outcome 'preempted' on termination.
[INFO] [WallTime: 1427740510.198243] Concurrent state 'pad_pause_nav' returned outcome 'preempted' on termination.
[INFO] [WallTime: 1427740510.199958] Concurrent state 'magnetic_strip' returned outcome 'preempted' on termination.
[INFO] [WallTime: 1427740510.201809] Concurrent Outcomes: {'stuck_on_carpet': 'preempted', 'magnetic_strip': 'preempted',
 'service_pause_nav': 'preempted', 'bumper': 'preempted', 'pad_pause_nav': 'preempted', 'NAV_SM': 'succeeded'}           
[INFO] [WallTime: 1427740510.202348] State machine terminating 'MONITORED_NAV':'succeeded':'succeeded'
[INFO] [WallTime: 1427740510.202641] SUCCEEDED
[INFO] [WallTime: 1427740510.376948] Waiting for services...
[INFO] [WallTime: 1427740510.383482] Done
Do move_base to ChargingPoint
[INFO] [WallTime: 1427740510.720509] State machine starting in initial state 'MONITORED_NAV' with userdata: 
        ['goal', 'result']
[INFO] [WallTime: 1427740510.721436] Concurrence starting with userdata: 
        ['goal']
[INFO] [WallTime: 1427740510.738248] State machine starting in initial state 'NAVIGATION' with userdata: 
        ['goal', 'n_nav_fails']
[ WARN] [1427740515.992480870]: Clearing costmap to unstuck robot (2.000000m).
[ WARN] [1427740516.192733228]: Clearing costmap to unstuck robot (0.000000m).
[ERROR] [1427740516.392619042]: Aborting because a valid plan could not be found. Even after executing all recovery behav
iors                                                                                                                     
[INFO] [WallTime: 1427740516.402160] State machine transitioning 'NAVIGATION':'planner_failure'-->'RECOVER_NAVIGATION
```

I think the issue might be in:

```
[INFO] [WallTime: 1427740503.177123] new node reached ChargingPoint
[INFO] [WallTime: 1427740503.177768] new node reached ChargingPoint          
```

 This maybe makes it try again even if after it was successful, and start with a "Do move_base..." thingy for some reason @Jailander ?

**Comment by Jailander** [comment]
I don't know what this could be, I'll commit the localise by topic thing and lets see if it happens with that

**Comment by bfalacerda** [comment]
This is still happening, even with https://github.com/strands-project/strands_navigation/pull/216. Here is what happens when I start navigation while docked, and tell the robot to go to ChargingPoint via the mdp action server:

```
[INFO] [WallTime: 1428328897.152912] Creating Reconfigure Client
[INFO] [WallTime: 1428328897.167949] new node reached ChargingPoint
[INFO] [WallTime: 1428328897.175348] new node reached ChargingPoint
[INFO] [WallTime: 1428328897.192880] All Done ...
Do move_base to ChargingPoint
[INFO] [WallTime: 1428328900.466217] State machine starting in initial state 'MONITORED_NAV' with userdata: 
        ['goal']
[INFO] [WallTime: 1428328900.466645] Concurrence starting with userdata: 
```

**Comment by bfalacerda** [comment]
solved by #216
